readmeの編集作業

narakatsu0622 opened the pull request:
#what
readmeを編集する
#why
データベース定義を追記するため

#アソシエーションの関係について記述しました。
#マークダウン記法のブラウザへの反映を確認しました。
#外部キーの型をpreference型に修正しました。
レビューお願いします。
#外部キー制約を追記しました。
#多対多のアソシエーションの定義をhas many trough に変更しました。

#中間テーブルの定義を追加しました。
#各テーブルの主キーのカラム名をidに変更しました。

#スペルミスの修正を行いました。
#アソシエーション定義を変更しました。

#アソシエーション定義を再変更しました。

#アソシエーションを変更、追記しました。
#中間テーブル名、それに付随するアソシエーションの変更を行った
#NOTNULL制約の変更
#外部キー項目の削除
＃データ型の変更
#deviceを使用する旨を追記

＃スペルミスの修正
#アソシエーションの変更

#テーブル定義の説明を変更
#idに関する記述を削除

#スペルミスの修正

narakatsu0622 committed:
read me edit

narakatsu0622 committed:
README edit

kenchan0503 commented on README.md:
```diff
@@ -1,24 +1,31 @@
+## groupsテーブル
+
+|colimn    |type   |index|null|unique|
+|group_id  |integer|y    |    |      |
```
外部キーはpreference型で書きましょう！

narakatsu0622 replied:
ご指摘ありがとうございます。
修正致します。

kenchan0503 commented:
wipの場合は何を確認してほしいかについてもお願いいたします（外し忘れであれば大丈夫です）

kenchan0503 commented:
アソシエーションの関係についての記述もお願いいたします

kenchan0503 commented:
マークダウンが効いていないようなのでご確認をお願いいたします

narakatsu0622 committed:
アソシエーションの定義

narakatsu0622 committed:
read_me_edit

#マークダウン記法のブラウザへの反映確認
#外部キーをpreference型へ修正しました。
#アソシエーションの定義を追加しました。

kenchan0503 commented on README.md:
```diff
@@ -1,24 +1,64 @@
+
+> Group has many Messages
+
+> Group has and belongs to many Users
```
has and でもいいのですがhas_many throughの方が一般的なのでそちらでお願いします！

narakatsu0622 replied:
わかりました。修正します。

narakatsu0622 replied:
修正しました。

kenchan0503 commented on README.md:
```diff
@@ -1,24 +1,64 @@
+| id      | integer   |      | y   | y     |
+| body    | text      |      | y   |       |
+| image   | string    |      | y   |       |
+| group_id| preference| y    |     |       |
```
preference型には外部キー制約もお願いします！

narakatsu0622 replied:
わかりました。修正します。

narakatsu0622 replied:
修正しました。

narakatsu0622 committed:
readme edit

アソシエーションの定義の記述変更
各データベース定義に対して外部キー制約の追記

kenchan0503 commented:
中間テーブルの内容が少し異なるので中間テーブルについて調べてみてください！

narakatsu0622 reviewed (commented):
了解しました。

narakatsu0622 committed:
readme edit

中間テーブルの追加
主キーのカラム名をidに統一

narakatsu0622 committed:
readme edit

中間テーブルのアソシエーションの修正

sato11 commented on README.md:
```diff
@@ -1,24 +1,78 @@
+###Usersテーブルに関するアソシエーション
+> User has many Messages
+>
+> User has many throuth Groups
```
中間テーブルのアソシエーションの記述が正確でないのでご確認ください。
またスペルミスにも気をつけましょう。

narakatsu0622 replied:
かしこまりました。
修正します。

sato11 commented on README.md:
```diff
@@ -1,24 +1,78 @@
+
+
+## Usersテーブル
+>
```
この引用符は必要ないと思うのですがいかがでしょう？
あってもなくても変わらないのであれば、後に残さないようにしましょう。

narakatsu0622 replied:
かしこまりました。
修正致します。

sato11 commented on README.md:
```diff
@@ -1,24 +1,78 @@
+| body    | text      |      | y   |       |
+| image   | string    |      | y   |       |
+| group_id| preference| y    |     |       |Groups.id
+| user_id | preference| y    |     |       |Users.id
```
preference というのがわからないのですが、ご自身で調べられましたでしょうか？

外部キーを指定する際に推奨されている型は references 型です。
http://qiita.com/sinagaki58/items/7edea51ef00e393834ca

メンターの勘違いやミスタイプでお手数おかけしてしまったのであれば申し訳ないのですが、基本的にこちらから指摘したことでも鵜呑みにせず、自身で咀嚼した上で反映させてくださりますと幸いです。

narakatsu0622 replied:
かしこまりました。
修正致します。

narakatsu0622 committed:
readme edit

#不要な記述の削除
#アソシエーションに関する記述の修正
#スペルミスの修正

narakatsu0622 committed:
readme edit

#スペルミス修正

narakatsu0622 committed:
readme edit

スペルミス修正

sato11 commented on README.md:
```diff
@@ -1,24 +1,76 @@
+###usersテーブルに関するアソシエーション
+> has_many : messages
+
+> has_and_belongs_to_many : groups
```
この記法でも問題ないといえば問題ないのですが、 `has_many :through` の方が後の拡張性が高いので推奨とされています。
今回は `has_many :through` を使って定義してみましょう！

narakatsu0622 replied:
わかりました。修正します！

narakatsu0622 replied:
修正しました。

narakatsu0622 committed:
readme edit

アソシエーション定義の変更

narukami894 commented on README.md:
```diff
@@ -1,24 +1,74 @@
+| user_id   |  references| y    |     |       |users.id
+
+### group_usersテーブルに関するアソシエーション
 
```
ここにもアソシエーションがあるはずですね！
has_manyに対応するものがあるはずです

narakatsu0622 replied:
ありがとうございます。修正いたします。

narukami894 commented on README.md:
```diff
@@ -1,24 +1,74 @@
+| colimn    | type      | index| null| unique|外部キー
+|-----------|-----------|------|-----|-------|-----
+| group_id  |  references| y    |     |       |group.id
+| user_id   |  references| y    |     |       |users.id
```
references型だと、null: falseになるのでは？

narukami894 commented on README.md:
```diff
@@ -1,24 +1,74 @@
+
+###Groupsテーブルに関するアソシエーション
+
+> has_many :messages, through: :users
```
messageはどれか一つのグループに属するはずです

narakatsu0622 replied:
ありがとうございます。修正いたします。

narukami894 commented on README.md:
```diff
@@ -1,24 +1,74 @@
+| colimn    | type       | index| null| unique|外部キー
+|-----------|------------|------|-----|-------|-----
+| id        | integer    | y    |     |       |
+| broup_name| text       |      |     |       |
```
ここはnameとかtitleとかのカラム名でいいと思います。
group_nameというカラム名をつけてしまうと、例えば値を取り出すときに
group.group_nameという風に書かなくてはいけなくなるので。

narukami894 replied:
あと、型はtextである必要はないですね。
textはかなり長い文字列にしか使わないので

narukami894 replied:
グループ名が空だと困るので、not null制約つけましょう

narakatsu0622 replied:
ありがとうございます。修正いたします。

narukami894 commented on README.md:
```diff
@@ -1,24 +1,74 @@
+| password| text   |      |     |       |
+
+###usersテーブルに関するアソシエーション
+> has_many : messages, through: :groups
```
has_many throughの理解が少し怪しいように見受けられるので、もう一度調べて、アソシエーションを書き直してみてください！

narakatsu0622 replied:
ありがとうございます。修正いたします。

narukami894 commented on README.md:
```diff
@@ -1,24 +1,74 @@
+| colimn  | type   | index| null| unique|外部キー
+|---------|--------|------|-----|-------|-----
+| id      | integer| y    |     | y     |
+| name    | text   |      |     | y     |
```
ユーザ名が空だと困るので、not null制約つけましょう

narukami894 commented on README.md:
```diff
@@ -1,24 +1,74 @@
+|---------|--------|------|-----|-------|-----
+| id      | integer| y    |     | y     |
+| name    | text   |      |     | y     |
+| mail    | text   |      |     | y     |
```
こちらもですね！
あと、deviseのデフォルトのカラム名だとmailではなくてemailですので、変更お願いいたします

narukami894 commented on README.md:
```diff
@@ -1,24 +1,74 @@
+| id      | integer| y    |     | y     |
+| name    | text   |      |     | y     |
+| mail    | text   |      |     | y     |
+| password| text   |      |     |       |
```
passwordもnot nullですね！

narukami894 commented on README.md:
```diff
@@ -1,24 +1,74 @@
+
+
+
+## usersテーブル
```
deviseを使用する旨を記述してください！

narakatsu0622 replied:
ありがとうございます。修正いたします。

narakatsu0622 replied:
修正しました。

narukami894 commented on README.md:
```diff
@@ -1,24 +1,74 @@
+
+|colimn   | type      | index| null| unique|外部キー|
+|---------|-----------|------|-----|-------|------|
+| id      | integer   |      | y   | y     |      |
```
idにnullは許可できないです！

narakatsu0622 replied:
ありがとうございます。修正いたします。

narukami894 commented on README.md:
```diff
@@ -1,24 +1,74 @@
+|colimn   | type      | index| null| unique|外部キー|
+|---------|-----------|------|-----|-------|------|
+| id      | integer   |      | y   | y     |      |
+| body    | text      |      | y   |       |
```
bodyもnot nullですね

narakatsu0622 replied:
ありがとうございます。修正いたします。

narukami894 commented on README.md:
```diff
@@ -1,24 +1,74 @@
+| id      | integer   |      | y   | y     |      |
+| body    | text      |      | y   |       |
+| image   | string    |      | y   |       |
+| group_id| references| y    |     |       |groups.id
```
referencesと書いてある時点で外部キーであることは自明なので、
外部キーという項目はそもそも不要かと思われます！

narakatsu0622 replied:
ありがとうございます。修正します。

narakatsu0622 committed:
readme edit

#外部キー項目の削除
#アソシエーションの変更、追記
#中間テーブル名、それに付随するアソシエーションの変更
#NOTNULL制約の変更
#外部キー項目の削除

narakatsu0622 committed:
readme edit

#行間の追加

narukami894 commented on README.md:
```diff
@@ -1,24 +1,88 @@
+| id      | integer   |      |     | y     |
+| body    | text      |      |     |       |
+| image   | text      |      | y   |       |
+| group_id| references| y    |     |       |
```
referencesは基本的にnull falseですね

narukami894 commented on README.md:
```diff
@@ -1,24 +1,88 @@
+
+
+## user_groupsテーブル
+
```
railsの命名規則では、中間テーブルはアルファベット順です！
この場合ですとgroup_usersテーブルになるはずです

narakatsu0622 replied:
ありがとうございます。修正します。

narukami894 commented on README.md:
```diff
@@ -1,24 +1,88 @@
+### usersテーブルに関するアソシエーション
+> has_many : messages,
+
+> has_many :groups, through => :user_groups
```
ハッシュロケット(=>)は使わないようにしましょう

narakatsu0622 replied:
ありがとうございます。修正します。

narukami894 commented on README.md:
```diff
@@ -1,24 +1,88 @@
+> has_many :messages
+
+> has_many :users, through: :user_groups
+
```
has_many 中間テーブル　を記述し忘れていますね

narakatsu0622 replied:
ありがとうございます。修正します。

narukami894 commented on README.md:
```diff
@@ -1,24 +1,88 @@
+| colimn    | type       | index| null| unique|
+|-----------|------------|------|-----|-------|
+| id        | integer    |      |     | y     |
+| neme      | string     |      |     |       |
```
スペルミスがいくつか見受けられます

narakatsu0622 replied:
ありがとうございます。修正します。

narakatsu0622 committed:
readme edit

#スペルミスの修正
#テーブル名の変更
#アソシエーションの変更、追記

narakatsu0622 committed:
readme edit

#アソシエーションの追記

narakatsu0622 committed:
readme edit

#不要な記述の削除

tetsu8dayo commented on README.md:
```diff
@@ -1,24 +1,90 @@
+
+|colimn   | type      | index| null| unique|
+|---------|-----------|------|-----|-------|
+| id      | integer   |      |     | y     |
```
idはnull許容でしょうか？
idは勝手に生成されるのでそもそも書かなくてもいいと思いますが、あえて書くならprimary key といったような記述も必要です。

tetsu8dayo commented on README.md:
```diff
@@ -1,24 +1,90 @@
+| image   | text      |      | y   |       |
+| group_id| references| y    |     |       |
+| user_id | references| y    |     |       |
+
```
group_id, user_idはnull許容でしょうか？

tetsu8dayo commented on README.md:
```diff
@@ -1,24 +1,90 @@
+
+| colimn  | type   | index| null| unique|
+|---------|--------|------|-----|-------|
+| id      | integer| y    |     | y     |
```
idにindexを貼るのはおかしい気がしますがそもそも自分で設定したりするのでしょうか？

tetsu8dayo replied:
同様にnull許容でしょうか？

tetsu8dayo replied:
idは勝手に生成されるので書かなくてもいいと思います。

tetsu8dayo commented on README.md:
```diff
@@ -1,24 +1,90 @@
+| colimn  | type   | index| null| unique|
+|---------|--------|------|-----|-------|
+| id      | integer| y    |     | y     |
+| name    | string | y    |     | y     |
```
nameはnull許容でしょうか？

tetsu8dayo commented on README.md:
```diff
@@ -1,24 +1,90 @@
+| name    | string | y    |     | y     |
+| mail    | text   |      |     | y     |
+| password| string |      |     |       |
+
```
こちらももう一度考え直しましょう。emailとpasswordはデバイスのデフォルトの設定で生成するならその旨を書いて、mailやpasswordに関する行は無くすべきだと思います。
もし書くならdeviseのデフォルトの設定を調べてその通りにしましょう

tetsu8dayo commented on README.md:
```diff
@@ -1,24 +1,90 @@
+|-----------|------------|------|-----|-------|
+| id        | integer    |      |     | y     |
+| name      | string     |      |     |       |
+
```
nameはnull許容でしょうか？

narakatsu0622 committed:
readme edit

#テーブルのid行を削除

narakatsu0622 committed:
readme edit

#テーブル定義からidの項目を削除
#index,null,uniqueに関する表現方法の変更

kenchan0503 commented on README.md:
```diff
@@ -1,24 +1,81 @@
+### usersテーブルに関するアソシエーション
+> has_many : messages,
+
+> has_many :groups, through: :user_groups
```
group_usersですね

kenchan0503 commented on README.md:
```diff
@@ -1,24 +1,81 @@
+> has_many :groups, through: :user_groups
+
+> has_many :user_groups
+
```
こちらも

kenchan0503 commented on README.md:
```diff
@@ -1,24 +1,81 @@
+
+> has_many :messages
+
+> has_many :users, through: :user_groups
```
こちらも

narakatsu0622 committed:
readme edit

#スペルミスの修正
